tinygrad/runtime/ops_clang.py

```diff
@@ -28,5 +59,24 @@ def __init__(self, name:str, lib:bytes):
 
   def __call__(self, *bufs, vals=(), wait=False): return cpu_time_execution(lambda: self.fxn(*bufs, *vals), enable=wait)
 
+class ClangJITProgram:
+  def __init__(self, name:str, lib:bytes):
+    mem = libc.mmap(None, len(lib), PROT_READ | PROT_WRITE | PROT_EXEC, MAP_ANON | MAP_PRIVATE | (MAP_JIT if OSX else 0), -1, 0)
+    if OSX:
+      global_handle.pthread_jit_write_protect_np(False)
+    ctypes.memmove(mem, lib, len(lib))
+    if OSX:
+      global_handle.pthread_jit_write_protect_np(True)
+      global_handle.sys_icache_invalidate(ctypes.c_void_p(mem), len(lib))
+    self.fxn = ctypes.cast(mem, ctypes.CFUNCTYPE(None))
+  def __call__(self, *bufs, vals=(), wait=False):
+    args = list(bufs) + list(vals)
+    # apple relaxes abi requirement for stack arguments to always be at least 8 byte aligned on arm64
+    # https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms
+    if platform.machine() == "arm64" and OSX: args = args[:8] + [ctypes.c_int64(a) if isinstance(a, int) else a for a in args[8:]]
```

Comment:
How does normal clang deal with this without this hack?

Reply:
Hmm, so the fix would require a macho parser?

Reply:
After looking into it, I'm okay with this ABI hack staying. Apple did something weird.